src/storage/mod.rs

```diff
@@ -1329,21 +1329,32 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
             callback
         );
 
-        if !self.enable_ttl && ttl != 0 {
-            return Err(Error::from(ErrorInner::TTLNotEnabled));
+        if !self.enable_ttl {
+            if ttls.iter().any(|&x| x != 0) {
+                return Err(Error::from(ErrorInner::TTLNotEnabled));
+            }
+        } else {
+            if ttls.len() != pairs.len() {
+                return Err(Error::from(ErrorInner::TTLsLenNotEqualPairs));
+            }
         }
```

Comment:
Maybe better to swap the order of then and else, because double negation is harder to understand.